src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

```diff
@@ -178,7 +168,9 @@ export function useSearchCurrencyLists() {
   const filterAssetsFromFavoritesBridgeAndAssetToSell = useCallback(
     (assets?: SearchAsset[]) =>
       filterAssetsFromBridgeAndAssetToSell(assets)?.filter(
-        curatedAsset => !favoritesList?.map(fav => fav.address).includes(curatedAsset.address)
+        curatedAsset =>
+          !favoritesList?.map(fav => fav.address).includes(curatedAsset.address) &&
+          !favoritesList?.map(fav => fav.address).includes(curatedAsset.mainnetAddress)
       ) || [],
     [favoritesList, filterAssetsFromBridgeAndAssetToSell]
```

Comment:
what about something like
```ts
!favoritesList?.some(({ address }) => curatedAsset.address === address && curatedAsset.mainnetAddress === address)
```
so we don't have to loop multiple times the same array